server/kolide/queries.go

```diff
@@ -64,3 +80,63 @@ type Query struct {
 	// table in the MySQL backend.
 	Packs []Pack `json:"packs" db:"-"`
 }
+
+const (
+	QueryKind = "Query"
+)
+
+type QueryObject struct {
+	ObjectMetadata
+	Spec QuerySpec `json:"spec"`
+}
+
+type QuerySpec struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Query       string `json:"query"`
+}
+
+func LoadQueriesFromYaml(yml string) ([]*Query, error) {
```

Comment:
are these yaml functions here because you already wrote them and you're saving them for the client implementation?

Reply:
Yes